crypto/bio/b_sock.c

```diff
@@ -387,6 +387,8 @@ int BIO_socket_wait(int fd, int for_read, time_t max_time)
     struct timeval tv;
     time_t now;
 
+    if (fd < 0 || fd >= FD_SETSIZE)
+        return -1;
     if (max_time == 0)
         return 1;
```

Comment:
likewise here,  what is so special with max_time==0 that this
returns 1, which means socket is readable?

Reply:
See above.
As written in the code comments of `BIO_connect_retry()`, there are some strange situations where also in the blocking case `BIO_do_connect()` returns with a retry condition, and for uniformity with the non-blocking case (with non-zero timeout/max_time) it is convenient to return with 1 (success) of `max_time == 0`.

Reply:
I do not understand what strange condition that might be.
BUT I think I don't like libcrypto calling into libssl (BIO_do_handshake)

Reply:
sorry mixed it up with SSL_do_handshake :-)